src/Sulu/Component/Content/Document/Behavior/RouteBehavior.php

```diff
@@ -31,4 +31,32 @@ public function getTargetDocument();
      * @param object $targetDocument
      */
     public function setTargetDocument($targetDocument);
+
+    /**
+     * Returns locale.
+     *
+     * @return string
+     */
+    public function getLocale();
+
+    /**
+     * Sets locale.
+     *
+     * @param string $locale
+     */
+    public function setLocale($locale);
```

Comment:
i have extracted an own behavior for this. also i have introduced an own document for the custom-url routes